### ISSUE TITLE
no-useless-initializer: false positive with type variables

### ISSUE THREAD

**Issue by ajafff** [fimbullinter/wotan#440]
```ts
function test<T, U extends string>(param: {t: T, u: U}) {
  let {t = '', u = ''} = param;
}
```

Currently has an error on `t = ''` and `u = ''`, but `t` could actually be undefined.
